jobqueue_features/clusters.py

```diff
@@ -306,7 +314,8 @@ def _get_maximum_scale(self, maximum_scale: int, default: int = 1) -> None:
         self.validate_positive_integer("maximum_scale")
 
     def _get_pure(self, pure: bool, default: Any = None) -> None:
-        self.pure = pure if pure is not None else default
+        if isinstance(pure, bool):
+            self.pure = pure if pure is not None else default
```

Comment:
The check against `None` was in condition `isinstance(pure, bool)`.

Reply:
Ok, I think what I actually want there is
```
if isinstance(default, bool):
    self.pure = pure if pure is not None else default
elif isinstance(pure, bool):
    self.pure = pure
```
Agree?

Reply:
I'm trying to avoid setting the attribute unless we were told to when the cluster was created, or we need it because we are in `mpi_mode`

Reply:
Ok, understood.